include/triton/Conversion/TritonGPUToLLVM/TargetInfoBase.h

```diff
@@ -10,6 +10,23 @@ class TargetInfoBase {
   virtual bool supportMaximumMinimum() const = 0;
   virtual Value callBallotOp(ConversionPatternRewriter &rewriter, Location loc,
                              Type type, Value cmp) const = 0;
+  virtual Value storeShared(ConversionPatternRewriter &rewriter, Location loc,
+                            Value ptr, Value val, Value pred) const = 0;
+  virtual Value loadShared(ConversionPatternRewriter &rewriter, Location loc,
+                           Value ptr, Type elemTy, Value pred) const = 0;
+  virtual Value shflSync(Location loc, ConversionPatternRewriter &rewriter,
+                         Value val, int i) const = 0;
+  virtual Value shflUpSync(Location loc, ConversionPatternRewriter &rewriter,
+                           Value val, int i) const = 0;
+  virtual Value shflIdxSync(Location loc, ConversionPatternRewriter &rewriter,
+                            Value val, int i) const = 0;
+  virtual Value shflIdxSync(Location loc, ConversionPatternRewriter &rewriter,
+                            Value val, Value i) const = 0;
+  virtual bool warpReduce(ConversionPatternRewriter &rewriter, Location loc,
+                          SmallVector<Value> &acc, triton::ReduceOp op,
+                          unsigned numLaneToReduce) const = 0;
+  virtual unsigned getShuffleIndex(triton::ReduceOp op, unsigned N,
```

Comment:
`numLaneToReduce` is only used to assert in amd backend

Reply:
the name is a bit confusing as shuffle isn't necessarily for reduction. Do you mean warp size?

Reply:
It's not really the warp size, its the offset between the two elements that are getting reduced, for example, `0x8` when you are reducing 16 elements.

Reply:
I'm not sure I understand the meaning for shuffle index then